com.unity.formats.fbx/Editor/Unity.Formats.Fbx.Editor.api

```diff
@@ -0,0 +1,125 @@
+// make sure the XML doc file is present and located next to the scraped dll
+namespace UnityEditor.Formats.Fbx.Exporter
+{
+    public static class ConvertToNestedPrefab
+    {
+        public static UnityEngine.GameObject Convert(UnityEngine.GameObject toConvert, string fbxDirectoryFullPath = default(string), string fbxFullPath = default(string), string prefabDirectoryFullPath = default(string), string prefabFullPath = default(string), UnityEditor.Formats.Fbx.Exporter.ConvertToPrefabSettingsSerialize exportOptions = default(UnityEditor.Formats.Fbx.Exporter.ConvertToPrefabSettingsSerialize));
+    }
+
+    public class ConvertToPrefabSettingsSerialize : UnityEditor.Formats.Fbx.Exporter.ExportOptionsSettingsSerializeBase
+    {
+        public virtual bool AllowSceneModification { get; }
+        public virtual bool ExportUnrendered { get; }
+        public virtual UnityEditor.Formats.Fbx.Exporter.LODExportType LODExportType { get; }
+        public virtual UnityEditor.Formats.Fbx.Exporter.Include ModelAnimIncludeOption { get; }
+        public virtual UnityEditor.Formats.Fbx.Exporter.ObjectPosition ObjectPosition { get; }
+        public ConvertToPrefabSettingsSerialize() {}
+    }
+
+    public enum ExportFormat
+    {
+        public const UnityEditor.Formats.Fbx.Exporter.ExportFormat ASCII = 0;
+        public const UnityEditor.Formats.Fbx.Exporter.ExportFormat Binary = 1;
+        public int value__;
+    }
+
+    public class ExportModelSettingsSerialize : UnityEditor.Formats.Fbx.Exporter.ExportOptionsSettingsSerializeBase
+    {
+        public virtual bool AllowSceneModification { get; }
+        public virtual bool EmbedTextures { get; }
+        public virtual bool ExportUnrendered { get; }
+        public virtual bool KeepInstances { get; }
+        public virtual UnityEditor.Formats.Fbx.Exporter.LODExportType LODExportType { get; }
+        public virtual UnityEditor.Formats.Fbx.Exporter.Include ModelAnimIncludeOption { get; }
+        public virtual UnityEditor.Formats.Fbx.Exporter.ObjectPosition ObjectPosition { get; }
+        public virtual bool PreserveImportSettings { get; }
+        public ExportModelSettingsSerialize() {}
+        public virtual bool Equals(object e);
+        public virtual int GetHashCode();
+        public void SetEmbedTextures(bool embedTextures);
+        public void SetExportUnredererd(bool exportUnrendered);
+        public void SetKeepInstances(bool keepInstances);
+        public void SetLODExportType(UnityEditor.Formats.Fbx.Exporter.LODExportType lodLevel);
+        public void SetModelAnimIncludeOption(UnityEditor.Formats.Fbx.Exporter.Include include);
+        public void SetObjectPosition(UnityEditor.Formats.Fbx.Exporter.ObjectPosition objectPosition);
+        public void SetPreserveImportSettings(bool preserveImportSettings);
+    }
+
+    public abstract class ExportOptionsSettingsSerializeBase : UnityEditor.Formats.Fbx.Exporter.IExportOptions
```

Comment:
This seems to add a lot of noise for the public API.
Would it make sense to make certain things only read and not write and have 2 factories for the 2 settings types ?

Reply:
@clusty : if I understand, you're proposing the public API be something like:
* Convert(ConvertSettings) with a public class ConvertSettings
* Export(ExportSettings) with a public class ExportSettings
The two classes are similar but have no relationship.

Then under the hood we can convert each settings class to the current overengineered hierarchy that we currently have.